File: php/shapes/Circle.php
<?php

//require_once "Shape.php";
//require_once "Quadrant.php";

class Circle extends Shape
{
    private float $radius;

    public function __construct(Quadrant $quadrant, float $radius)
    {
        parent::__construct($quadrant);
        $this->radius = $radius;
    }

    public function isInShape(float $x, float $y): bool
    {
        return $this->isInQuadrant($x, $y) && ($x ** 2 + $y ** 2 < ($this->radius ** 2));
    }
}

File: src/main/java/com/moleus/web/service/areaCheck/shapes/Circle.java
package com.moleus.web.service.areaCheck.shapes;

import com.moleus.web.service.areaCheck.quadrant.Point;
import com.moleus.web.service.areaCheck.quadrant.Quadrant;

public class Circle extends AbstractShape {
    private final float radius;

    public Circle(Quadrant quadrant, float radius) {
        super(quadrant);
        this.radius = radius;
    }

    @Override
    public boolean isInShape(Point point) {
        return super.isInQuadrant(point) &&
            point.x() * point.x() + point.y() * point.y() <= this.radius * this.radius;
    }
}
